File: week3/eunbeen/소수찾기.java
import java.util.*;

class Solution {
    public Set<Integer> set = new HashSet();
    String[] tracking = {};
    boolean[] check = {};
    public int solution(String numbers) {
        int answer = 0;
        
        tracking = numbers.split("");
        check = new boolean[tracking.length];
        
        permutation(0, "");
        
        Iterator<Integer> iter = set.iterator();
        while(iter.hasNext()){
            var num = iter.next();
            if(num == 2) answer += 1;
            answer += num%2 != 0 ? isPrime(num) : 0;
        }
        
        return answer;
    }
    private int isPrime(int num) {
        if(num <= 1) return 0;
        for(int i=2; i <= Math.sqrt(num); i++) {
            if(num%i==0) return 0;
        }
        return 1;
    }
    private void permutation(int index, String sum) {
        if (!sum.isEmpty()) {
            set.add(Integer.parseInt(sum));
        }
        
        for (int i = 0; i < tracking.length; i++) {
            if (!check[i]) {
                check[i] = true;
                permutation(index++, sum + tracking[i]);
                check[i] = false;
            }
        }
    }
}

File: week3/lee/소수찾기.py
import itertools
def check(n): # 소수 체크
    if n <= 1:
        return False
    if n <= 2:
        return True
    else:
        for i in range(2,n):
            if n%i == 0:
                return False

    return True

def solution(numbers):

    answer = []
    for j in range(1,len(numbers)+1):
        per = list(map(''.join, itertools.permutations(numbers,j))) # 1~끝까지 뽑기
        per = set(per) # 중복제거
        for i in per:
            p = int(i) # 앞에 0제거
            if check(p):
                answer.append(p)
    answer = set(answer)
    return len(answer)

print(solution("011"	))